api-inference-community/docker_images/spacy/tests/test_api_text_classification.py

```diff
@@ -0,0 +1,134 @@
+import json
+import os
+from unittest import TestCase, skipIf
+
+from app.main import ALLOWED_TASKS
+from starlette.testclient import TestClient
+from tests.test_api import TESTABLE_MODELS
+
+
+@skipIf(
+    "text-classification" not in ALLOWED_TASKS,
+    "text-classification not implemented",
+)
+class TextClassificationTestCase(TestCase):
+    def setUp(self):
+        model_id = TESTABLE_MODELS["text-classification"]
+        self.old_model_id = os.getenv("MODEL_ID")
+        self.old_task = os.getenv("TASK")
+        os.environ["MODEL_ID"] = model_id
+        os.environ["TASK"] = "text-classification"
+        from app.main import app
+
+        self.app = app
+
+    def tearDown(self):
+        if self.old_model_id is not None:
+            os.environ["MODEL_ID"] = self.old_model_id
+        else:
+            del os.environ["MODEL_ID"]
+        if self.old_task is not None:
+            os.environ["TASK"] = self.old_task
+        else:
+            del os.environ["TASK"]
+
+    def test_simple(self):
+        inputs = "It is a beautiful day outside"
+
+        with TestClient(self.app) as client:
+            response = client.post("/", json={"inputs": inputs})
+        self.assertEqual(
+            response.status_code,
+            200,
+        )
+        content = json.loads(response.content)
+        self.assertEqual(type(content), dict)
+        self.assertEqual(
```

Comment:
I think this unit test is too specific to the specific model. I would suggest to just look at the types and lengths of the response instead of checking all the keys.

Reply:
Once it's more generic, it would also be great to have this in `docker_images/common/tests/test_api_text_classification`. That way future libraries will have a template for `text-classification` tests and your impact will be multiplied  🚀

Reply:
I am asserting that the length of the list must be equal to 28 in the case of ""explosion/en_textcat_goemotions"". Is the same okay for `docker_images/common/tests/test_api_text_classification`

Reply:
@osanseviero Hi! I have made the changes, please let me know about any further changes.